pkg/dfns-api-client/auth_transport.go

```diff
@@ -0,0 +1,278 @@
+package dfns_api_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github/dfns/dfns-sdk-go/pkg/credentials"
+)
+
+const (
+	UserActionHeader    = "x-dfns-useraction"
+	appIDHeader         = "x-dfns-appid"
+	nonceHeader         = "x-dfns-nonce"
+	authorizationHeader = "authorization"
+	contentTypeHeader   = "Content-Type"
+)
+
+type authTransport struct {
+	*DfnsApiOptions
+}
+
+func (auth *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	auth.setHeaders(req)
+
+	performUserAction, err := shouldPerformUserAction(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if performUserAction {
+		if err := auth.performUserActionRequest(req); err != nil {
+			return nil, err
+		}
+	}
+
+	return executeRequest(req)
+}
+
+func shouldPerformUserAction(req *http.Request) (bool, error) {
+	if ok := slices.Contains([]string{"POST", "PUT", "DELETE"}, req.Method); !ok {
+		return false, nil
+	}
+
+	authHeaderStr := req.Header.Get(UserActionHeader)
+	if authHeaderStr == "" {
+		return true, nil
+	}
+	defer req.Header.Del(UserActionHeader)
```

Comment:
nit: maybe you rather clean this header on line 34, before `if performUserAction {` ? So that this header also gets cleaned if it's a GET and someone set it to `true` manually ? And feel like deleting a header should not be part of a function called `shouldPerformUserAction`